From: Kill Your TV <killyourtv@i2pmail.org>
Date: Wed, 25 May 2011 00:26:10 +0300
Subject: rename jcpuid

---
 core/c/jcpuid/build.sh |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

--- a/core/c/jcpuid/build.sh
+++ b/core/c/jcpuid/build.sh
@@ -39,8 +39,8 @@
 *kFreeBSD*|Linux*)
 	COMPILEFLAGS="-fPIC -Wall"
 	INCLUDES="-I. -Iinclude -I$JAVA_HOME/include -I$JAVA_HOME/include/linux"
-	LINKFLAGS="-shared -Wl,-soname,libjcpuid-x86-linux.so"
-	LIBFILE="lib/freenet/support/CPUInformation/libjcpuid-x86-linux.so";;
+	LINKFLAGS="-shared -Wl,-soname,libjcpuid.so"
+	LIBFILE="lib/freenet/support/CPUInformation/libjcpuid.so";;
 esac
 
 echo "Compiling C code..."
