Add debian package vesion to the EXTRA version field

--- a/router/java/src/net/i2p/router/RouterVersion.java
+++ b/router/java/src/net/i2p/router/RouterVersion.java
@@ -21,7 +21,7 @@
     public final static long BUILD = 0;
 
     /** for example "-test" */
-    public final static String EXTRA = "";
+    public final static String EXTRA = "deb1";
     public final static String FULL_VERSION = VERSION + "-" + BUILD + EXTRA;
     public static void main(String args[]) {
         System.out.println("I2P Router version: " + FULL_VERSION);
